### PR TITLE
Makers Monday Pull req

### PR DESCRIPTION
**Pull request by mikey-roberts**

### CODE REVIEW

File: lib/rules.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
+module Rules
+
+  MAXIMUM_FRAMES = 10
```
Comment (coo990) [inline lib/rules.rb:3]: Good use of constant variables

File: lib/rules.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  def pins_hit?
+    both_throws != 0
+  end
+  
+  def gutter?
+    both_throws == 0
+  end
+  
+  def both_throws
+    first_throw + second_throw
```
Comment (coo990) [inline lib/rules.rb:26]: Good definitions of first_throw and second_throw

File: lib/rules.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  # Max validation
+
+  def single_turn_valid(bowl)
+    bowl <= MAXIMUM_PINS
+  end
+
+  def both_turns_valid
+    both_throws <= MAXIMUM_PINS
```
Comment (coo990) [inline lib/rules.rb:36]: Single responsibility methods

File: lib/scorecard.rb [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  frame == 10 && first_throw == strike?
+end
+
+def user_input
+  loop do
+      puts "Please enter your first throw"
+    @first_throw = STDIN.gets.chomp.to_i
+    break if strike? && single_turn_valid(first_throw)
+      puts "Please enter your second throw"
+    @second_throw = STDIN.gets.chomp.to_i
```
Comment (coo990) [inline lib/scorecard.rb:31]: Good conversion of user input into an integer

File: lib/scorecard.rb [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+  def score_hit(first_throw, second_throw)
+    # pins_per_frame << [first_throw, second_throw]
+    first_throw + second_throw
+  end
+
+
+  private 
+
+  def strike_last_turn?
+    @previous_turn_bonus.last == "strike"
```
Comment (coo990) [inline lib/scorecard.rb:88]: Simple use for "strike"

File: spec/rules_spec.rb [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    end
+
+    it 'strike is set as 10 pins being hit' do
+      expect(strike.strike?).to eq 10
+    end
+
+    it 'spare is set at 10 pins being hit in total across two turns' do
+      expect(spare.spare?).to eq 10
+    end
+
```
Comment (coo990) [inline spec/rules_spec.rb:26]: Good testing for spare edge cases, you could probably encapsulate the spare tests into one.

File: spec/scorecard_spec.rb [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    let(:subject) { described_class.new }
+    let(:first_throw) { double(:first_throw, first_throw: 10) }
+    let(:second_throw) { double(:second_throw, second_throw: 7) }
+
+    describe '#initialize' do
+      it 'initializes the class' do
+        expect(subject.frame).to eq 1
+        expect(subject.first_throw).to eq 0
+        expect(subject.second_throw).to eq 0
+        expect(subject.bonus_roll).to eq 1
```
Comment (coo990) [inline spec/scorecard_spec.rb:17]: Good initialise test

File: spec/scorecard_spec.rb [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        expect(subject.pins_per_frame).to eq []
+        expect(subject.total_score).to eq 0
+        expect(subject.previous_turn_bonus).to eq []
+      end
+    end
+
+    describe '#user_input' do
+      it 'returns valid input' do 
+        subject.stub(:user_input).and_return(10)
```
Comment (coo990) [inline spec/scorecard_spec.rb:26]: Good test for user input.

File: spec/scorecard_spec.rb [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    describe '#set_turn_bonus' do
+      it 'sets bonus based off the number of strikes' do
+      expect(subject.set_turn_bonus(10, 10)).to eq ["strike"]
+      end
+    end
+
+    describe '#score_hit' do
+      it 'calculates score' do 
+        subject.score_hit(5, 5)
+        expect(subject.score_hit(5, 5)).to eq 10
```
Comment (coo990) [inline spec/scorecard_spec.rb:41]: Nice and simple to test score hit

**Review by coo990 — COMMENTED** [review]
Really good code, like the different methods and your rules module to keep the program in check.
